chain/jsonrpc/src/lib.rs

```diff
@@ -188,6 +188,9 @@ impl JsonRpcHandler {
     }
 
     async fn process_request(&self, request: Request) -> Result<Value, RpcError> {
+        let _rpc_processing_time =
```

Comment:
Have you checked that an unused variable gets dropped at the end of scope since I was under impression that Rust does not guarantee it. You should have an explicit `drop` to make sure that the lifetime covers the whole block.

Reply:
Unused variables named `_` are dropped immediately but otherwise they are dropped at the end of the scope. I have checked that.